### PR TITLE
Solutions for Chapter 4 - Partial

### PR DESCRIPTION
**Pull request by piushkumar**
### Solutions for Chapter 4 - Partial

cc @vrom911 @chshersh

### CODE REVIEW

File: src/Chapter4.hs [hunk 1]
```diff
@@ -293,7 +301,8 @@ values and apply them to the type level?
 -}
 instance Functor (Secret e) where
     fmap :: (a -> b) -> Secret e a -> Secret e b
-    fmap = error "fmap for Box: not implemented!"
+    fmap f (Reward a) = (Reward (f a))
```
Comment (vrom911) [inline src/Chapter4.hs:304]: Parenthesis redundant here 🙂 
```suggestion
    fmap f (Reward a) = Reward (f a)
```

File: src/Chapter4.hs [hunk 2]
```diff
@@ -472,10 +486,11 @@ Implement the Applicative instance for our 'Secret' data type from before.
 -}
 instance Applicative (Secret e) where
     pure :: a -> Secret e a
-    pure = error "pure Secret: Not implemented!"
+    pure = Reward
 
     (<*>) :: Secret e (a -> b) -> Secret e a -> Secret e b
-    (<*>) = error "(<*>) Secret: Not implemented!"
+    (Reward f) <*> a = fmap f a
```
Comment (vrom911) [inline src/Chapter4.hs:492]: Nice work on reusing `Functor` instance for `Secret`! :world_map:

**Review by vrom911 — APPROVED** [review]
Great start!
I f you have any questions or concerns about the following task, just ping us, we are here to help and discuss difficult topics 🙂

**Comment by piushkumar** [comment]
> 
> 
> Great start!
> I f you have any questions or concerns about the following task, just ping us, we are here to help and discuss difficult topics 🙂

@vrom911  yes taking time to get my head around usage of functor and Monads. Finding them super hard to understand and utilize. Do you have any links to read / understand more about them ?

**Comment by vrom911** [comment]
No worries, these are new concepts so it is okay that it would take some time to actually see the usages for that.

There are few resources where you can read, I can recommend this one, as it describes it quite nicely:
http://learnyouahaskell.com/functors-applicative-functors-and-monoids

**Comment by chshersh** [comment]
Thanks for participating in Learn4Haskell! Hope you enjoyed this Hacktoberfest 🥳 

I am going to close this PR as we can't merge the solutions into our course 🔐

**Pull request closed without merging**